packages/client/src/pages/CreateSafe.js

```diff
@@ -101,6 +105,34 @@ function CreateSafe({ web3 }) {
     createTreasury,
   } = web3;
   const [safeOwners, setSafeOwners] = useState([{ name: "", address }]);
+  const [safeOwnersValidByAddress, setSafeOwnersValidByAddress] = useState({});
+
+  const checkSafeOwnerAddressesValidity = async (newSafeOwners) => {
+    const newSafeOwnersValidByAddress = {};
+
+    for (const so of newSafeOwners) {
+      const isAddress = isAddr(so.address);
+
+      if (isAddress) {
+        try {
+          await injectedProvider.account(so.address);
+          newSafeOwnersValidByAddress[so.address] = true;
+        } catch (err) {
+          newSafeOwnersValidByAddress[so.address] = false;
+        }
+      } else {
+        newSafeOwnersValidByAddress[so.address] = false;
+      }
+    }
+
+    setSafeOwnersValidByAddress(newSafeOwnersValidByAddress);
+  };
+
+  const onSafeOwnersChange = (newSafeOwners) => {
+    setSafeOwners(newSafeOwners);
+    // skip first safe owner since it's the connected user and won't have an address
+    checkSafeOwnerAddressesValidity(newSafeOwners.slice(1));
+  };
```

Comment:
can we move this into a hook or reducer so it can be shared in other places that need address validation? such as in the transfer tokens flow

Reply:
makes sense, moved into a `useAddressValidation` hook